database/scrapers/building_scrapper.ts

```diff
@@ -0,0 +1,158 @@
+import { DateTime } from "luxon";
+import * as fs from "node:fs/promises";
+import { Readable } from "node:stream";
+
+import { BaseScraperModule, TaskHandle } from "#commands/db_scrape";
+import { BuildingIcon } from "#enums/building_icon";
+import Building from "#models/building";
+import Campus from "#models/campus";
+import FilesService from "#services/files_service";
+
+const buildingsPath = "https://admin.topwr.solvro.pl/items/Buildings";
+const assetsPath = "https://admin.topwr.solvro.pl/assets/";
+
+interface BuildingsData {
+  id: number;
+  name: string;
+  latitude: number;
+  longitude: number;
+  addres: string; //intentional typo
+  cover: string | null;
+  food: boolean | null;
+  naturalName: string;
+  externalDigitalGuideIdOrURL: string;
+  externalDigitalGuideMode: string;
+  createdAt: string;
+  updatedAt: string;
+}
+
+interface CampusesData {
+  data: {
+    name: string;
+    cover: string;
+    buildings: string[];
+  }[];
+}
+
+export default class BuildingScrapper extends BaseScraperModule {
+  name = "building and campuses scrapper";
+  description =
+    "scrapes pwr buildings data from directus and campuses from local file: './assets/campuses.json'";
+
+  private filesService = new FilesService();
+
+  async run(task: TaskHandle) {
+    try {
+      this.logger.info("starting reading campuses file...");
+      const campusesData = JSON.parse(
+        await fs.readFile("./assets/campuses.json", { encoding: "utf-8" }),
+      ) as CampusesData | undefined;
+      if (campusesData === undefined) {
+        throw new Error("failed reading ./assets/campuses.json");
+      }
```

Comment:
MDN docs for `JSON.parse` don't mention `undefined` as a possibility, but other JS types are possible.
I'd check if the result is an `object` and throw an error if it's not (mentioning what the actual type was)